platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/preconsensus/PreconsensusEventFileManager.java

```diff
@@ -622,4 +628,20 @@ private void updateFileSizeMetrics() {
                     .set(((double) totalFileByteCount) / files.size() * UnitConstants.BYTES_TO_MEBIBYTES);
         }
     }
+
+    /**
+     * Performs changes to the stream when a software upgrade occurs.
+     *
+     * @param softwareUpgrade true if a software upgrade has occurred
+     * @throws UncheckedIOException if the required changes on software upgrade cannot be performed
+     */
+    public void onSoftwareUpgrade(final boolean softwareUpgrade) {
```

Comment:
I feel like its better if the PCES has no idea about the software upgrade. We should just call clear on it if there is one, to separate concers

Reply:
I agree, I think it would be cleaner if the `if (softwareUpgrade && clearOnSoftwareUpgrade)` logic happened in the caller's scope.